### ISSUE TITLE
"Auto Dark Mode for Web Contents" improvement suggestions

### ISSUE THREAD

**Issue by thataboy** [Alex313031/thorium#500]
**Is your feature request related to a problem? Please describe.**
The experimental flag "Auto Dark Mode for Web Contents" is great for people like me who are light sensitive and want all web pages to be dark. However, it does have a couple of drawbacks:

- it often breaks web pages, making elements unreadable, such as light text on light background or dark text on dark background, or making icons disappear
- there's a performance hit. It is still quite faster than using an extension (which is why I prefer it) but it's a performance hit nevertheless; how much depends on the web page.

**Describe the solution you'd like, including relevant patches or source**

Some possible improvements:

- bypass this feature if web page already has dark theme. I don't know how this can be done exactly. Does the browser need to analyze the css and detect if dark theme in present, or do web sites have some way to indicate it has a dark theme either by default or available on demand
- allow toggling this feature on/off on the fly. The ideal would be to maintain a list of exceptions, or blacklist / whitelist.

**Comment by Alex313031** [comment]
@thataboy The best thing is to not select "Enable" but rather, "Enabled with selective inversion of non-image elements"

**Comment by thataboy** [comment]
> @thataboy The best thing is to not select "Enable" but rather, "Enabled with selective inversion of non-image elements"

Yes I already do this. I've tried all the options, actually. They have the pros and cons, but none works all the time. Hence my suggestion for on the fly toggle.

BTW, the latest chromium beta build seems to have removed this feature entirely. I do hope Thorium doesn't follow suit. That would be really unfortunate.

**Comment by GuelBandeira** [comment]
allowing toggling this feature on/off would be such a improvement, this is a thing that is missing for me, because I am used to Opera GX that has this feature, allowing me to toggle the force dark elements on/off when I want with the right-click button, and now on Thorium I am missing that
